test/banned_commands.bats

```diff
@@ -23,6 +23,7 @@ banned_commands=(
     # Process substitution isn't POSIX compliant and cause trouble
     "<("
     # source isn't POSIX compliant. . behaves the same and is POSIX compliant
+    # Except in fish, where . is deprecated, and will be removed in the future.
     source
```

Comment:
This is interesting to know, thanks @maxnordlund Do you have suggestions for how we should handle this in future once `.` is removed in Fish?

Reply:
Just keep using `source` I guess? It's only `.` that's deprecated, which might not be that obvious now that I read it again.

Reply:
Our plan was to deprecate use of `source` internally because `.` is more POSIX compliant. If Fish is dropping `.`, then this will affect all usage of `.` in our codebase.

Reply:
Fish isn't a POSIX shell at all, so I think it would be better to keep them separated. But most of asdf is written in bash, right? There it would be fine to enforce `.` over `source`.

If you want to go full POSIX, you could switch to `#!/bin/sh` and see what shellcheck finds. 🤔

Reply:
I think it is a bit soon for us to go "FUll POSIX"

Reply:
@jthegedus we could just ignore `.fish` files in our banned commands script. That would probably be the best way to handle since fish isn't POSIX.

Reply:
Ah, of course! Yes, sounds like a good solution.